apps/nextjs-website/src/helpers/metadata.helpers.ts

```diff
@@ -1,27 +1,48 @@
 import { translations } from '@/_contents/translations';
-import { ResolvedMetadata } from 'next';
-import { AbsoluteTemplateString } from 'next/dist/lib/metadata/types/metadata-types';
-import { Twitter } from 'next/dist/lib/metadata/types/twitter-types';
+import { MakeMetadataFunction } from '@/lib/types/makeMetadataFunction';
+import { Metadata } from 'next';
 
-export const getTitleFromMarkdown = (markdown: string): string => {
-  const lines = markdown.split('\n').filter((line) => line !== '');
-  const titles = lines.filter((line) => line.startsWith('#'));
-  const title = titles[0]?.replace('# ', '').trim() ?? '';
-  return title;
+export const makeMetadata: MakeMetadataFunction = ({
+  parent,
+  title,
+  description,
+  url,
+  image,
+}) => {
+  const { shared } = translations;
+  const previousTitle = parent?.title?.absolute || shared.siteTitle;
+  const metadataTitle = title ? `${previousTitle} | ${title}` : previousTitle;
+
+  return {
+    title: metadataTitle,
+    description: description || '',
+    url: url || '',
+    openGraph: getOpenGraphMetadata(metadataTitle, description, image),
+    twitter: getTwitterMetadata(metadataTitle, description, image),
+  };
 };
 
-export const getTwitterMetadata = (title: string): Twitter => ({
+const getOpenGraphMetadata = (
+  title: string,
+  description?: string,
+  image?: string
+): Metadata['openGraph'] => ({
+  title,
+  type: 'website',
+  locale: 'it_IT',
+  description: description || '',
+  images: image,
+});
+
+const getTwitterMetadata = (
+  title: string,
+  description?: string,
+  image?: string
+): Metadata['twitter'] => ({
   title,
+  description: description || '',
```

Comment:
Having an empty string or an `undefined` or `null` value is the same in term of meta tags, I'd keep the empty string because some social networks could write the url in the description field and it could not be the best solution to show it. Would you agree?